docs/docsite/rst/dev_guide/developing_program_flow_modules.rst

```diff
@@ -464,20 +454,75 @@ this. If a module has to use this on its own, it should instantiate an
         facility = getattr(syslog, facility_name, syslog.LOG_USER)
         syslog.openlog(NAME, 0, facility)
 
+For more details, see the :ref:`DEFAULT_SYSLOG_FACILITY` configuration option.
+
 .. versionadded:: 2.1
 
+
 _ansible_version
 ^^^^^^^^^^^^^^^^
 
-This parameter passes the version of Ansible that runs the module. To access
-it, a module should instantiate an `AnsibleModule` and then retrieve it
-from :attr:`AnsibleModule.ansible_version`. This replaces
-:attr:`ansible.module_utils.basic.ANSIBLE_VERSION` from
-:ref:`module_replacer`.
+This argument passes the version of Ansible to the module. To access it, a module should instantiate the ``AnsibleModule`` method and then retrieve the version from :attr:`AnsibleModule.ansible_version`. This replaces :attr:`ansible.module_utils.basic.ANSIBLE_VERSION` from :ref:`module_replacer`.
 
 .. versionadded:: 2.1
 
 
+_ansible_module_name
+^^^^^^^^^^^^^^^^^^^^
+
+Type: ``str``
+
+This argument passes the information to modules about their name. For more details see, the configuration option :ref:`DEFAULT_MODULE_NAME`.
+
+
+_ansible_string_conversion_action
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+This argument provides instructions about what modules should do after the values of the user-specified module parameters are converted to strings. For more details, see the :ref:`STRING_CONVERSION_ACTION` configuration option.
+
+
+_ansible_keep_remote_files
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Type: ``bool``
+
+This argument provides instructions that modules must be ready if they need to keep the remote files. For more details, see the :ref:`DEFAULT_KEEP_REMOTE_FILES` configuration option.
+
+
+_ansible_socket
+^^^^^^^^^^^^^^^
+This argument provides modules with a socket for persistent connections. The argument is created using the :ref:`PERSISTENT_CONTROL_PATH_DIR` configuration option.
+
+
+_ansible_shell_executable
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Type: ``bool``
+
+This argument ensures that modules use the designated shell executable. For more details, see the :ref:`ansible_shell_executable` remote host environment parameter.
+
+
+_ansible_tmpdir
+^^^^^^^^^^^^^^^
+
+Type: ``str``
+
+This argument provides instructions to modules that all commands must use the designated temporary directory, if created.
+
+You can configure a part of this argument such as a root path. However, based on the circumstances the end directory is generated randomly for each task, or host in one of these:
+
+* :ref:`DEFAULT_LOCAL_TMP`
+* `remote_tmp <https://docs.ansible.com/ansible/latest/collections/ansible/builtin/sh_shell.html#parameter-remote_tmp>`_
+* `system_tmpdirs <https://docs.ansible.com/ansible/latest/collections/ansible/builtin/sh_shell.html#parameter-system_tmpdirs>`_
+
+As a result, using the ``ansible.cfg`` configuration file to activate or customize this setting will not guarantee that you control the full value.
```

Comment:
There's a public interface that should probably be recommend using over either of the internal tmp dir parameters: https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/basic.py#L538.

Mentioning that the tmp dir is randomized is helpful, but it probably doesn't need to be the focus. Maybe this section could be modified along the lines of:

```
The temporary directory designated by the action plugin. Modules should access this parameter by using the public ``tmpdir`` property, which will create a temporary directory if the action plugin didn't set the parameter.

The directory name is generated randomly, and the the root of the directory is determined by one of these:

... links to config options ...
```